docs/custom-filters.adoc

```diff
@@ -36,7 +42,49 @@ For common things like logging and metric facade APIs it is recommended to use t
 == Protocol filters
 
 A protocol filter is a `public` top-level, concrete class with a particular public constructor and which implements
-one or more protocol filter interfaces.
+one or more protocol filter interfaces. You can implement 3 distinct types of Custom Protocol Filter:
+
+- <<Specific Message Protocol Filters>>
+- <<Request/Response Protocol Filters>>
+- <<Composite Filters>>
```

Comment:
I separated CompositeFilter so it is now described as a convenience and is not mentioned as a "type of Custom Protocol Filter", but I did keep it up here where we are talking filter interfaces and have a cross-reference from the configuration section. What do you think? Invert it so that it's documented in the configuration section and have a pointer from up here to there?